### PR TITLE
Feature: Replace standard bloc with cubit

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -37,8 +36,8 @@ class HomePage extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    context.read<AuthBloc>().add(const AuthEventInitialize());
-    return BlocConsumer<AuthBloc, AuthState>(
+    context.read<AuthCubit>().initialize();
```
Comment (utpal-barman) [inline lib/main.dart]: let's not call init() in the build method, build methods can be frequently rebuilt

call it when a cubit it created

File: lib/main.dart [hunk 2]
```diff
@@ -37,8 +36,8 @@ class HomePage extends StatelessWidget {
...
+    return BlocConsumer<AuthCubit, AuthState>(
```
Comment (utpal-barman) [inline lib/main.dart]: we can optimize consumer or listener with `listenWhen`, which will reduce the number of listening

File: lib/services/auth/bloc/auth_cubit.dart [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:mynotes/services/auth/bloc/auth_state.dart';
+import 'package:mynotes/services/auth/firebase_auth_provider.dart';
+
+class AuthCubit extends Cubit<AuthState> {
```
Comment (utpal-barman) [inline lib/services/auth/bloc/auth_cubit.dart]: let's introduce dependency injection by annotation, but let's introduce it in separate PR

File: lib/services/auth/bloc/auth_cubit.dart [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    final user = authProvider.currentUser;
+    if (user == null) {
+      emit(const AuthStateLoggedOut(
+        exception: null,
+        isLoading: false,
+      ));
+    } else if (!user.isEmailVerified) {
+      emit(const AuthStateNeedsVerification(isLoading: false));
+    } else {
+      emit(AuthStateLoggedIn(user: user, isLoading: false));
```
Comment (utpal-barman) [inline lib/services/auth/bloc/auth_cubit.dart]: sealed classes are good, but we have some limitations, we can introduce a status property and make a union class to handle loading loaded success failure, this is how we are maintaining the projects

File: lib/services/auth/bloc/auth_cubit.dart [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    }
+  }
+
+  void register(String email, String password) async {
+    try {
+      await authProvider.createUser(
```
Comment (utpal-barman) [inline lib/services/auth/bloc/auth_cubit.dart]: to follow DDD, use only usecases in cubits , NO other service / providers etc

File: lib/views/login_view.dart [hunk 6]
```diff
@@ -57,59 +56,140 @@ class _LoginViewState extends State<LoginView> {
...
-                        .add(const AuthEventShouldRegister());
-                  },
-                  child: const Text('Not registered yet? Register here!'),
-                ),
-              ],
+                  const SizedBox(height: 10),
+                  SizedBox(
+                    height: 55,
+                    width: double.infinity,
+                    child: TextFormField(
```
Comment (utpal-barman) [inline lib/views/login_view.dart]: we always prefer reusable widgets rather than writing big chunk of codes

File: lib/views/register_view.dart [hunk 7]
```diff
@@ -69,18 +68,13 @@ class _RegisterViewState extends State<RegisterView> {
...
-                      ),
-                    );
+                context.read<AuthCubit>().register(email, password);
               },
               child: const Text('Register'),
             ),
             TextButton(
               onPressed: () {
-                context.read<AuthBloc>().add(const AuthEventLogOut());
+                context.read<AuthCubit>().logOut();
```
Comment (utpal-barman) [inline lib/views/register_view.dart]: if a same cubit's method is required in the same widget, consider usign a final 
```dart
final cubit = context.read<AuthCubit>();
....
cubit.register(email, password);
....
onPressed: cubit.logout;

```

File: lib/views/verify_email_view.dart [hunk 8]
```diff
@@ -23,17 +22,15 @@ class _VerifyEmailViewState extends State<VerifyEmailView> {
               "If you haven't received a verification email yet, press the button below"),
           TextButton(
             onPressed: () async {
-              context
-                  .read<AuthBloc>()
-                  .add(const AuthEventSendEmailVerification());
+              context.read<AuthCubit>().sendEmailVerification();
```
Comment (utpal-barman) [inline lib/views/verify_email_view.dart]: can be written like this:

```dart
 onPressed: context.read<AuthCubit>().sendEmailVerification;
```

**Review by utpal-barman — APPROVED** [review]
Looks good! 🚀

File: lib/services/auth/bloc/auth_cubit.dart (hunk 5 above)
Comment (ghost) [reply lib/services/auth/bloc/auth_cubit.dart]: will introduce usecases soon.

**Pull request merged**